src/main/java/org/jenkinsci/plugins/workflow/actions/ArgumentsAction.java

```diff
@@ -0,0 +1,169 @@
+                filteredArguments.put(entry.getKey(), entry.getValue());
+            }
+        }
+        return filteredArguments;
+    }
+
+    /**
+     * Get just the fully stored, non-null arguments
+     * This means the arguments with all {@link NotStoredReason} or null values removed
+     * @param n FlowNode to get arguments for
+     * @return Map of all completely stored arguments
+     */
+    @Nonnull
+    public static Map<String, Object> getFilteredArguments(@Nonnull FlowNode n) {
+        ArgumentsAction act = n.getPersistentAction(ArgumentsAction.class);
+        return act != null ? act.getFilteredArguments() : Collections.EMPTY_MAP;
+    }
+
+    /** Return a tidy string description for the step arguments, or null if none is present or we can't make one
+     *  See {@link StepDescriptor#argumentsToString(Map)} for the rules
+     */
+    @CheckForNull
+    public static String getStepArgumentsAsString(@Nonnull FlowNode n) {
+        if (n instanceof StepNode) {
+            StepDescriptor descriptor = ((StepNode) n).getDescriptor();
+            if (descriptor != null) {  // Null if plugin providing descriptor was uninstalled
+                Map<String, Object> filteredArgs = getFilteredArguments(n);
+                return descriptor.argumentsToString(filteredArgs);
+            }
+        }
+        return null;  // non-StepNode nodes can't have step arguments
+    }
+
+    /**
+     * Return a fast view of internal arguments, without creating immutable wrappers
+     * @return Internal arguments
+     */
+    @Nonnull
+    protected abstract Map<String, Object> getArgumentsInternal();
+
+    /**
+     * Get the value of a argument, or null if not present/not stored.
+     * Use {@link #getArgumentValueOrReason(String)} if you want to return the {@link NotStoredReason} rather than null.
+     * @param argumentName Argument name of step to look up.
+     * @return Argument value or null if not present/not stored.
+     */
+    @CheckForNull
+    public Object getArgumentValue(@Nonnull String argumentName) {
+        Object val = getArgumentValueOrReason(argumentName);
+        return (val == null || val instanceof NotStoredReason) ? null : val;
+    }
+
+    /**
+     * Get the argument value or its {@link NotStoredReason} if it has been intentionally omitted.
+     * @param argumentName Name of step argument to find value for
+     * @return Argument value, null if nonexistent/null, or NotStoredReason if it existed by was masked out.
+     */
+    @CheckForNull
+    public Object getArgumentValueOrReason(@Nonnull String argumentName) {
+        return getArgumentsInternal().get(argumentName);
+    }
+
+    /**
+     * Test if {@link Step} arguments are persisted in an unaltered form.
+     * @return True if full arguments are retained, false if some have been removed for security, size, or other reasons.
+     */
+    public abstract boolean isFullArguments();
```

Comment:
Any reason you cannot provide a default implementation? (i.e. a transient `Boolean` field if `null` just walk `getArgumentsInternal().values()` looking for a `NotStoredReason` followed by an idempotent write to the boolean field... or just the walk of `getArgumentsInternal().values()` looking for a `NotStoredReason`)

If there is a reason why you cannot provide a default implementation using the walk then you probably want to advertise that reason in the javadoc so that implementations do not fall foul of the issue you are concerned with